File: ruby/user_delete.rb
require 'faraday'
require 'json'

#
# Delete a User from an Account
#

# Required API Variables
require './vars.rb'

# Set API method variables
user_id = 1 # User ID to destroy /
            # You cannot destroy your own account

# URL and Path
url = "https://#{DOMAIN}.gimlet.us"
path = "/api/v1/users/#{user_id}.json"

# New connection with base URL and path
conn = Faraday.new(url: url)              # New connection with base URL
conn.basic_auth(EMAIL, API_KEY)           # Set the authentication header
response = conn.delete(path)              # DELETE API request

# Print the results
results = JSON.parse(response.body)
puts "RESULTS: #{results.inspect}"

File: php/user_delete.php
<?php
require './vars.php';

// Set API method variables
$user_id = 1; // User ID to delete
              // You cannot destroy your own account

$api_url = "https://{$DOMAIN}.gimlet.us/api/v1/users/{$user_id}.json";

$req = curl_init($api_url);
$credentials = "{$EMAIL}:{$API_KEY}";
curl_setopt($req, CURLOPT_USERPWD, $credentials);
curl_setopt($req, CURLOPT_CUSTOMREQUEST, "DELETE");
curl_setopt($req, CURLOPT_RETURNTRANSFER, true);

$data = curl_exec($req);

// Do what you need with the data
var_dump($data);

// ?>
